Source/Core/Core/State.cpp

```diff
@@ -388,32 +371,28 @@ static void CompressAndDumpState(CompressAndDumpState_args& save_args)
 
   if (header.size != 0)  // non-zero header size means the state is compressed
   {
-    lzo_uint i = 0;
+    u32 bytes_compressed = 0;
+
     while (true)
     {
-      lzo_uint32 cur_len = 0;
-      lzo_uint out_len = 0;
-
-      if ((i + IN_LEN) >= buffer_size)
-      {
-        cur_len = (lzo_uint32)(buffer_size - i);
-      }
-      else
-      {
-        cur_len = IN_LEN;
-      }
+      u32 bytes_to_compress = std::min((u32)LZ4_MAX_INPUT_SIZE, header.size - bytes_compressed);
+      char* out = new char[bytes_to_compress];
```

Comment:
Also please no raw new/delete.

Reply:
https://github.com/lz4/lz4/blob/dev/lib/lz4.h#L215-L224C7

Reply:
I did see that bound function, but given that our size is originally an u32, if it exceeds the signed int limit it will fail the bounds check and it will output 0.

Reply:
I guess if it outputs 0 then we know to set the limit to `LZ4_MAX_INPUT_SIZE`